### ISSUE TITLE
[CP] Copying fallback font family is unstable

### ISSUE THREAD

**Issue by IchordeDionysos** [flutter/flutter#118446]
### issue_link

https://github.com/flutter/flutter/issues/118392

### Commit Hash

[ddad6f1](https://github.com/flutter/flutter/commit/ddad6f16352f3a103329d2531505bd2c87a7462a)

### Target

beta

### pr_link

https://github.com/flutter/flutter/pull/118466

### Impacted Users

Packages copying package fonts

### Impact Description

Copying/Applying `TextStyle` with `fontFamilyFallback` will result in bugs and fallback fonts not being applied.

This is due to the fact that while copying the property will not stay the same:
`style.copyWith() != style`

### Workaround

Change this code snippet:

```dart
const style = TextStyle(fontFamilyFallback: <String>['Roboto'], package: 'a');
final copiedStyle = style.copyWith(); // copiedStyle != style
```

and explicitly specify the `fontFamilyFallback` when using `copyWith` or `apply` like this:

```
const style = TextStyle(fontFamilyFallback: <String>['Roboto'], package: 'a');
final copiedStyle = style.copyWith(fontFamilyFallback: <String>['Roboto']); // copiedStyle == style
```

But the behavior is unexpected and may lead to hidden/undetected bugs in an undetermined number of projects/packages.

### Risk

low

### Test Coverage

yes

### Validation Steps

The change is covered with a dedicated unit test.

**Comment by itsjustkevin** [comment]
@IchordeDionysos could you open a cherry-pick pr to the beta branch (flutter-3.7-candidate.1) with this change and update the PR link?

Once done, I will approve this cherry-pick for the queue.

CC @jason-simmons @mdebbar

**Comment by IchordeDionysos** [comment]
@itsjustkevin I've updated it with the PR to `flutter-3.7-candidate.1`

I was confused to which branch to open the PR to as there are multiple candidate branches
<img width="177" alt="image" src="https://user-images.githubusercontent.com/10195482/212393281-b00b763b-e035-4790-ab62-33225fd88da7.png">

And the cherry-pick form describes a different format for the branches `release-caniddate-branch.version`:
<img width="849" alt="image" src="https://user-images.githubusercontent.com/10195482/212393413-863ac8a9-059e-470f-a9ff-818e85c29641.png">

**Comment by IchordeDionysos** [comment]
Ahh now I got it 👌

**Comment by CaseyHillers** [comment]
This was shipped in the 3.7.0-1.5 beta
